src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+      for( XYCoord ownedFac : factoryOwnership.keySet() )
+      {
+        final Commander owner = factoryOwnership.get(ownedFac);
+        if( owner == null )
+          continue; // Don't barf in weird maps
+
+        final Unit inf = new Unit(owner, owner.getUnitModel(UnitModel.TROOP));
+        inf.x = ownedFac.xCoord;
+        inf.y = ownedFac.yCoord;
+
+        final GamePath infPath = findFeasiblePath(inf, propXYC, map);
+        if( null == infPath || infPath.getPathLength() < 1 )
+          continue; // Can't reach this city
+
+        int oldDistance = Integer.MAX_VALUE;
+        if( possibleOwners.containsKey(owner.army) )
+          oldDistance = possibleOwners.get(owner.army);
+
+        int distance = infPath.getFuelCost(inf, map) * 100 / inf.getMovePower(map);
+        if( distance < oldDistance )
+          possibleOwners.put(owner.army, distance);
+      }
+
+
+      // Calculate who's the closest, and if that army has real competition for this prop
+      Army closestArmy = null;
+      int closestDistance = Integer.MAX_VALUE;
+      for( Army army : possibleOwners.keySet() )
+      {
+        int distance = possibleOwners.get(army);
+        if( distance < closestDistance )
+        {
+          closestArmy = army;
+          closestDistance = distance;
+        }
+      }
+
+      boolean contested = false;
+      for( Army army : possibleOwners.keySet() )
+      {
+        if(!army.isEnemy(closestArmy))
+          continue;
+        int distance = possibleOwners.get(army);
+        int distanceDelta = Math.abs(closestDistance - distance);
+        contested |= ( distanceDelta <= 100 );
+      }
+
+      // If it isn't contested and we want to try to cap it, add it to the list
+      if( contested )
+        contestedProps.add(propXYC);
+      else if( viewer == closestArmy
+            && viewer.isEnemy(map.getLocation(propXYC).getOwner()) ) // Don't try to cap it if we already own it
+        rightfulProps.add(propXYC);
+    }
+
+    // Build cap chains to factories; don't continue them, since cap chains from that factory will be considered separately
+    ArrayList<ArrayList<CapStop>> factoryCapChains = new ArrayList<>();
+    while (!rightfulFactories.isEmpty())
+    {
+      final XYCoord dest = rightfulFactories.remove(0);
+      Utils.sortLocationsByDistance(dest, startingFactories);
+      final XYCoord start = startingFactories.get(0);
+      final Commander owner = factoryOwnership.get(start);
+
+      final Unit inf = new Unit(owner, owner.getUnitModel(UnitModel.TROOP));
+      inf.x = start.xCoord;
+      inf.y = start.yCoord;
+
+      final GamePath infPath = findFeasiblePath(inf, dest, map);
+      if( null == infPath || infPath.getPathLength() < 1 )
+        continue; // Can't reach
+
+      int distance = infPath.getFuelCost(inf, map);
+
+      ArrayList<CapStop> chain = new ArrayList<>();
+      CapStop build = new CapStop(start);
+      build.extraTurns = distance/inf.getMovePower(map) - 13;
```

Comment:
minus 13? Also it still seems odd to require a map to find a unit's move power.

Reply:
Minus 13 because that boosts the "assumed profit" from the cap chain, so it happens first.
This is necessary because cap chains to grab factories end at those factories (the cap chains from that factory are considered separately), and one property's income isn't superfantastic.

Reply:
As for the move power, you could argue either way.
The `extraTurns` metric won't be perfectly accurate in all cases with CO jank (since it doesn't plan the exact individual moves).
OTOH, it may give a better estimate, depending on the exact jank in question.

In summary, I'm hedging against my own future poor choices :v

Reply:
Why not just prioritize factory caps explicitly? I guess because a "sufficiently far away" factory should (probably) be worth less than a bunch of nearby cities? I just like to avoid fudge factors when possible.
I see that it matches `CapStopFundsComparator.turnLimit` - are those intentionally the same, or coincidentally (and otherwise unrelated)?

Reply:
> Why not just prioritize factory caps explicitly?

Why have two lists when I can get away with one? Laziness.
> I guess because a "sufficiently far away" factory should (probably) be worth less than a bunch of nearby cities?

Uhhhh. I hadn't considered that at all. It's not wrong, but claiming that as my reasoning would be a lie.
> I just like to avoid fudge factors when possible.

Valid. This is supposed to be a big enough "fudge" that you'd expect to see effectively two sub-lists within the list.
> I see that it matches `CapStopFundsComparator.turnLimit` - are those intentionally the same, or coincidentally (and otherwise unrelated)?

Coincidence, probably. The gremlin that is my brain really likes it when I type `13` and `42`. You might have noticed this tendency.

Reply:
OK. So factories are just assumed to be 13 turns closer to being captureable because reasons. Gotcha. I'd (as usual) advocate for adding a comment to explain what you are doing, since it's pretty nonobvious.

> Coincidence, probably. The gremlin that is my brain really likes it when I type 13 and 42. You might have noticed this tendency.

Not sure I have? Huh. Good thing I don't have anything against fudge factors. Or magic numbers. Or magic fudge factors.

Reply:
That's fair.

There are 66 instances of `13` in our source, and 16 of `42`.